pandas/core/indexes/base.py

```diff
@@ -5217,7 +5221,7 @@ def get_indexer_non_unique(self, target):
         tgt_values = target._get_engine_target()
 
         indexer, missing = self._engine.get_indexer_non_unique(tgt_values)
-        return ensure_platform_int(indexer), missing
+        return ensure_platform_int(indexer), ensure_platform_int(missing)
```

Comment:
perhaps you could just comment on why this change is necessary, and what it does?